fabric_cf/actor/core/policy/network_node_inventory.py

```diff
@@ -121,29 +117,28 @@ def __check_component_labels_and_capacities(self, *, available_component: Compon
 
         # Checking capacity for component
         capacity_delegations = available_component.get_capacity_delegations()
-        for delegation_id, delegation_list in capacity_delegations.items():
-            for delegated in delegation_list:
-                if Pool.ispoolmention(delegated) or Pool.ispooldefinition(delegated):
-                    # ignore pool mentions and definitions for now
-                    continue
-                self.logger.debug(f"available_capacity_delegations : {delegated} {type(delegated)} for component {available_component}")
-                delegated_capacity = Capacities().set_fields(**delegated)
-                requested_component.capacity_allocations = delegated_capacity
-                break
+        delegation_id, deleg = capacity_delegations.get_sole_delegation()
+        self.logger.debug(f"Available capacity_delegations: {deleg} {type(deleg)} format {deleg.get_format()}")
+        # ignore pool definitions and references for now
+        if deleg.get_format() != DelegationFormat.SinglePool:
+            return None
+        # get the Capacities object
+        delegated_capacity = deleg.get_details()
+        self.logger.debug(f"available_capacity_delegations : {capacity_delegations} {type(capacity_delegations)} "
+                          f"for component {available_component}")
+        requested_component.capacity_allocations = delegated_capacity
 
         # Check labels
         # FIXME this isn't quite right - we will need to pick one of available labels, but OK to leave for now
         label_delegations = available_component.get_label_delegations()
-        for delegation_id, delegation_list in label_delegations.items():
-            for delegated in delegation_list:
-                if Pool.ispoolmention(delegated) or Pool.ispooldefinition(delegated):
-                    # ignore pool mentions and definitions for now
-                    continue
-                self.logger.debug(f"available_label_delegations : {delegated} {type(delegated)} for component "
-                                  f"{available_component}")
-                delegated_label = Labels().set_fields(**delegated)
-                requested_component.label_allocations = delegated_label
-                break
+        delegation_id, deleg = capacity_delegations.get_sole_delegation()
```

Comment:
fabric_cf/actor/core/policy/network_node_inventory.py: Line 134
This should be changed to label_delegations.get_sole_delegation(). I think this is causing the Assertion failure in UpdateNode when updating ASM in FimHelper.

Reply:
Oh yeah. D'oh. Must be cut-and-paste artifact of some sort.